src/hooks/UrqlUseQuery.re

```diff
@@ -0,0 +1,52 @@
+[@bs.deriving abstract]
+type useQueryStateJs('a) = {
+  fetching: bool,
+  [@bs.optional] data: 'a,
+  [@bs.optional] error: UrqlCombinedError.t
+};
+
+[@bs.deriving abstract]
+type useQueryArgs('a) = {
+  query: string,
+  [@bs.optional] variables: 'a,
+  [@bs.optional] requestPolicy: UrqlTypes.requestPolicy,
+  [@bs.optional] pause: bool
+}
+
+type partialOperationContextFunc = option(UrqlTypes.partialOperationContext) => unit;
```

Comment:
I'd say let's just define this type inline on L17 below (right in the tuple as the second argument) since it's a somewhat unique one-off.